src/routes/session.ts

```diff
@@ -4,26 +4,69 @@ import {
 	getUser,
 	comparePassword,
 } from '../platform/auth';
-import { sbvrUtils } from '../platform';
+import { resinApi, root, sbvrUtils } from '../platform';
 import { captureException, handleHttpErrors } from '../platform/errors';
 import { resetCounter } from '../lib/rate-limiting';
 import { RequestHandler } from 'express';
 import { SetupOptions } from '..';
 
 const { BadRequestError, NotFoundError } = sbvrUtils;
 
-export const whoami: RequestHandler = (req, res) =>
-	getUser(req)
-		.then(user =>
-			res.send({ id: user.id, username: user.username, email: user.email }),
-		)
-		.catch(err => {
-			if (handleHttpErrors(req, res, err)) {
-				return;
+export const whoami: RequestHandler = async (req, res) => {
+	try {
+		let user: AnyObject = await getUser(req);
+
+		// If the user email is available, then the record was successfully
+		// populated by the passport middleware
+		if (user.email == null) {
+			// If we get to this point, then the request is most likely being made
+			// using a scoped access token.
+			// First we check to see if this token can access the user resource,
+			// then if it can, we retrieve user document that corresponds to the
+			// `actor` value on the token using root privileges.
+			const results = await resinApi.get({
```

Comment:
Why do we retrieve the matching resource with root privileges?  Doesn't the fact that this query succeeds mean that the token itself could fetch them directly, ie:
```suggestion
			const result = await resinApi.get({
				resource: 'user',
				passthrough: { req },
				options: {
					$filter: {
						actor: user.actor,
					},
					$select: ['id', 'username', 'email'],
				},
			});
			if (result) {
				user = result
			}

Reply:
@Page- we can't simulate the `/whoami` response for users with scoped access tokens without fetching by root if an implementation blacklists the email field, which is what we do on balenaCloud.

Reply:
Right, and there's ways to bypass the whitelist for internal requests but open api wouldn't know how to do that